src/test/java/swinglib/JButtonBuilderTest.java

```diff
@@ -0,0 +1,70 @@
+package swinglib;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import java.awt.event.ActionEvent;
+import java.util.Arrays;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import javax.swing.JButton;
+
+import org.hamcrest.core.Is;
+
+import org.junit.Test;
+
+public class JButtonBuilderTest {
+
+  @Test
+  public void checkTitle() {
+    final String value = "testing title";
+    final JButton button = JButtonBuilder.builder()
+        .title(value)
+        .actionListener(() -> {
+        })
+        .build();
+    assertThat(button.getText(), Is.is(value));
+  }
+
+  @Test
+  public void checkActionListener() {
+    // button action will be to add one to our integer, we'll fire the button action and verify we get the +1
+    final AtomicInteger integer = new AtomicInteger(0);
+    final JButton button = JButtonBuilder.builder()
+        .title("title")
+        .actionListener(integer::incrementAndGet)
+        .toolTip("toolTip")
+        .build();
+
+    Arrays.stream(button.getActionListeners())
+        .forEach(listener -> listener.actionPerformed(new ActionEvent(new Object(), 0, "")));
+    assertThat(integer.get(), Is.is(1));
+
+    assertThat(button.getToolTipText(), Is.is("toolTip"));
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void titleIsRequired() {
```

Comment:
Quick question about these types of tests...  Is the intention to ensure `build()` throws an exception if the title is anything other than a non-empty string?  Or if `title()` should throw an exception under the same circumstances?

If the former, then this test might be passing for the wrong reason.  It appears that `title()` is throwing `IllegalArgumentException`, and the test never calls `build()`.  Would it be more correct to just not call `title()` in this case to ensure `build()` flags the unspecified required property?  (The code in `build()` looks like it only checks to see if the field is non-`null`, so it ends up throwing `NullPointerException` if the `title()` call is omitted.)

If the latter, then could the test be simplified to just `JButtonBuilder.builder().title("")`?

Maybe both cases need tests?